src/common/kube-helpers.ts

```diff
@@ -16,16 +16,50 @@ function resolveTilde(filePath: string) {
   return filePath;
 }
 
+function loadFromFile(file: string) {
+  return loadFromString(fse.readFileSync(file, 'utf8'));
+}
+
+function loadFromString(config: string) {
+  try {
+    const obj = yaml.safeLoad(config);
+
+    // Validate the context structure for invalid context objects
+    if ('contexts' in obj) {
+      for (const [i, value] of obj["contexts"].entries()) {
+        if (!value.name || !value.context.cluster || !value.context.user) {
+          logger.error(`loadFromString: context ${JSON.stringify(value)} in kubeconfig has an error so will not be made available to Lens`);
+          obj["contexts"].splice(i, 1);
+        }
+      }
+    }
+    return yaml.dump(obj);
+  } catch (err) {
+    const error = String(err);
+    logger.error(`Could not pre-validate contexts key in supplied config string - ${config}`)
+    return config;
```

Comment:
This just punts to `kc.loadFromString()` in `loadConfig()`, which is ok because it is the same behaviour as before adding pre-validation but just wondering if we should throw here instead?

Reply:
@jim-docker - I have now wrapped the original loadConfig logic in a try/catch so we can actually catch errors from the Kubernetes Client functions (this wasn't happening before). So, my thought process was I only want to pre-validate the context object and if I can't do that because of some error we should just return the original config and let the Kubernetes Client create a kubeconfig object and catch the errors from that. 

If I catch and re-throw an error then we won't generate a kubeconfig object and right now Lens won't even load up!